calculator/dom.js

```diff
@@ -0,0 +1,160 @@
+let buttons = document.querySelectorAll('.button'),
+    numbers = document.querySelectorAll('.number'),
+    operations = document.querySelectorAll('.operation'),
+    decimal = document.getElementById('decimal'),
+    cleaner_all = document.getElementById('ac'),
+    cleaner_window = document.getElementById('c'),
+    display = document.getElementById('display'),
+    sign = document.getElementById('sign'),
+    memoryCurrentNumber = 0,
+    memoryNewNumber = false,
+    memoryError = false,
+    memoryPendingOperation = '';
+
+for (let i = 0; i < numbers.length; i++) {
```

Comment:
maybe instead of creating so much listeners, you can benefit from even bubbling? catch the events on body and handle the logic based on target?